docs/plugins/best_practices.md

```diff
@@ -0,0 +1,175 @@
+    end-user to choose which one they want. If they installed napari with `pip
+    install napari[all]`, then the `[all]` extra will (currently) install
+    `PyQt5` for them from pypi.  If they installed via `conda install napari`,
+    then they'll have `PyQt5`, but via anaconda cloud instead of pypi. Lastly,
+    they may have installed napari with PySide2
+
+    Here's what can go wrong if you *also* declare one of these backends:
+
+    - If they installed via `conda install napari` and then they install your
+      plugin via `pip` (or via the builtin plugin installer, which currently
+      uses `pip`), then there *will* be a binary incompatibility between the
+      their conda `pyqt` installation, and the new pip "`PyQt5`" installation.
+      *This will very likely lead to a broken environment*. This is an
+      unfortunate consequence of [package naming
+      decisions](https://github.com/ContinuumIO/anaconda-issues/issues/1554),
+      and it's not something napari can fix.
+    - Alternatively, they may end up with *both* PyQt and PySide in their
+      environement, and while that's not always an environment "death sentence",
+      it too can lead to unexpected and difficulty to debug problems.
+
+- **Don't import from PyQt5 or PySide2 in your plugin: use `qtpy`.**
+
+    If you use `from PyQt5 import QtCore` (or similar) in your plugin, but the
+    end-user has chosen to use `PySide2` for their Qt backend — or vice versa —
+    then your plugin will fail to import.  Instead use `from qtpy import
+    QtCore`.  `qtpy` is a [Qt compatibility
+    layer](https://github.com/spyder-ide/qtpy) that will import from whatever
+    backend is installed in the environment.
+
+- **Try not to depend on packages that require C compilation, but do not offer "wheels".**
+
+    ````{tip}
+    This requires some awareness of how your dependencies are built and distributed...
+
+    Some python packages write a portion of their code in lower level languages like C or C++ and compile that code into "C Extensions" that can be called by python at runtime.  This can *greatly* improve performance, but it means that the package must be compiled for *each* platform (i.e. Windows, Mac, Linux) that the package wants to support.  Some packages do this compilation step ahead of time, by distributing "[wheels](https://realpython.com/python-wheels/)"
+    on [PyPI](https://pypi.org/)... or by providing pre-compiled packages via `conda`.  Other packages simply distribute the source code (as an "sdist") and expect the end-user to compile it on their own computer.  Compiling C code requires software that is not always installed on every computer. (If you've ever tried to `pip install` a package and had it fail with a big wall of red text saying something about `gcc`, then you've run into a package that doesn't distribute wheels, and you didn't have the software required to compile it).
```

Comment:
eek... I thought about it, but I don't really want to "highlight" a specific package or plugin that did something we're discouraging

Reply:
ooops sorry, I totally agree there. I meant, an example of a dependency that requires compiling C code - I think `pytorch` is mentioned further down?

My reasoning was that if there's some common dependencies that do this, it might pay to highlight a few of them to show that it's not just very esoteric packages you might depend on.

Reply:
yeah, but torch [provides wheels](https://pypi.org/project/torch/#files) ... so it's not a great example.  The package that I first saw that had a dependency that most definitely didn't provide wheels but definitely required compilation was [probreg](https://pypi.org/project/probreg/) (but that kinda outs a specific napari plugin)

Reply:
Ah ok easy. Let's just leave it as is then. Thanks @tlambert03